src/land/smap_ssm2ioda.py

```diff
@@ -149,6 +161,42 @@ def _read(self):
         DimDict['Location'] = len(self.outdata[('dateTime', 'MetaData')])
 
 
+def get_observation_time(filename, ncd, vals):
+    # get observation time from file if present fallback to extraction from filename
+    times = np.empty_like(vals, dtype=np.int64)
+    if 'tb_time_seconds' in ncd.variables.keys():
+        refsec = ncd.groups['Soil_Moisture_Retrieval_Data'].variables['tb_time_seconds'][:].ravel()
+        for i, isec in enumerate(refsec):
+            dt = j2000_base_date + timedelta(seconds=isec)
+            times[i] = round((dt - epoch).total_seconds())
```

Comment:
```suggestion
            times = [round(((j2000_base_date + timedelta(seconds=isec)) - epoch).total_seconds()) for isec in ncd.groups['Soil_Moisture_Retrieval_Data'].variables['tb_time_seconds'][:].ravel()]
```

untested but any advantage?